### DISCUSSION TITLE
Thank you for this Project!

### DISCUSSION THREAD

**Discussion by JustDr00py** [shorepine/tulipcc#150] [General]
First off Thank you for such an amazing firmware for the T-Deck. Im absolutely in love with using your software on my T-Deck. being able to code an esp32 on the go is nice. Thanks for including ESP-NOW as well it makes a nice addition to the stack.

I'm curious what are the limitations? Could we add a sim module and use it to SMS?

**Comment by bwhitman** [comment]
I've never tried, but I don't see a reason you can't add a SIM module like this one - https://circuitdigest.com/microcontroller-projects/interfacing-sim800l-module-with-esp32

It looks like it connects over a UART. On the T-Deck, you may be able to use the existing UART pins on the back top left there. There's a UART module for Micropthon that could communicate with it. I've never tried this and it may require some changes to Tulip itself, but it's certainly possible!

**Reply by JustDr00py** [reply]
Thanks for the reply, I was not able to use the grove port to connect uart to the sim module yet only because I don't have the wire for it, but I was able to successfully send an sms by using espnow and relaying the message to the sim module connected to another esp32. So it sort of works as a portable sms device. Thanks again for an amazing project.
